src/Psalm/Internal/Codebase/ClassLikes.php

```diff
@@ -675,7 +676,23 @@ public function classImplements(string $fq_class_name, string $interface): bool
         }
         $class_storage = $this->classlike_storage_provider->get($fq_class_name);
 
-        return isset($class_storage->class_implements[$interface_id]);
+        if (isset($class_storage->class_implements[$interface_id])) {
+            return true;
+        }
+
+        foreach (array_keys($class_storage->class_implements) as $implementing_interface_lc) {
+            if (!isset($this->classlike_aliases[$implementing_interface_lc])) {
+                continue;
+            }
+
+            $aliased_interface_lc = strtolower($this->classlike_aliases[$implementing_interface_lc]);
```

Comment:
It's probably better to use `ClassLikes::getUnAliasedName()` rather than access `$classlike_aliases` directly. It would likely make fixing #6330 easier.